okapi-core/src/main/java/org/folio/okapi/util/CorsHelper.java

```diff
@@ -39,20 +39,17 @@ public static void addCorsHandler(Router router, final TenantManager tenantManag
       String meth = ctx.request().getHeader(HttpHeaders.ACCESS_CONTROL_REQUEST_METHOD);
       HttpMethod method = meth != null ? HttpMethod.valueOf(meth) : ctx.request().method();
       String moduleId = ctx.request().getHeader(XOkapiHeaders.MODULE_ID);
-      tenantManager.get(tenantId)
-          .onFailure(cause -> HttpResponse.responseError(ctx, 400, cause.getMessage()))
-          .onSuccess(tenant -> {
-            ModuleCache moduleCache = tenantManager.getModuleCache(tenant);
-            List<ModuleInstance> list = moduleCache.lookup(newPath, method, moduleId);
-            for (ModuleInstance mi : list) {
-              if (mi.isHandler() && mi.getRoutingEntry().isDelegateCors()) {
-                ctx.data().put(DELEGATE_CORS, true);
-                ctx.data().put(DELEGATE_CORS_MODULE_INSTANCE, mi);
-                break;
-              }
-            }
-            ctx.next();
-          });
+      Tenant t = new Tenant();
+      t.getDescriptor().setId(tenantId);
+      List<ModuleInstance> list = tenantManager.getModuleCache(t).lookup(newPath, method, moduleId);
```

Comment:
Let's leave things as they are (mostly) for now.. Can you make getModuleCache take a string rather than a Tenant class? That will make things a little prettier. Thx

Reply:
> Let's leave things as they are (mostly) for now.. Can you make getModuleCache take a string rather than a Tenant class? That will make things a little prettier. Thx

Done.